[2024-06-14T19:28:39.614+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: s3_to_ml_to_postgres.setup_create_table manual__2024-06-14T19:27:22.099468+00:00 [queued]>
[2024-06-14T19:28:39.618+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: s3_to_ml_to_postgres.setup_create_table manual__2024-06-14T19:27:22.099468+00:00 [queued]>
[2024-06-14T19:28:39.618+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 1
[2024-06-14T19:28:39.622+0000] {taskinstance.py:2191} INFO - Executing <Task(RedshiftDataOperator): setup_create_table> on 2024-06-14 19:27:22.099468+00:00
[2024-06-14T19:28:39.625+0000] {standard_task_runner.py:60} INFO - Started process 963 to run task
[2024-06-14T19:28:39.627+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 's3_to_ml_to_postgres', 'setup_create_table', 'manual__2024-06-14T19:27:22.099468+00:00', '--job-id', '25', '--raw', '--subdir', 'DAGS_FOLDER/ml_dataset_preparation.py', '--cfg-path', '/tmp/tmpcg7ggpj_']
[2024-06-14T19:28:39.628+0000] {standard_task_runner.py:88} INFO - Job 25: Subtask setup_create_table
[2024-06-14T19:28:39.645+0000] {task_command.py:423} INFO - Running <TaskInstance: s3_to_ml_to_postgres.setup_create_table manual__2024-06-14T19:27:22.099468+00:00 [running]> on host 22dc78ccc862
[2024-06-14T19:28:39.675+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='s3_to_ml_to_postgres' AIRFLOW_CTX_TASK_ID='setup_create_table' AIRFLOW_CTX_EXECUTION_DATE='2024-06-14T19:27:22.099468+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-06-14T19:27:22.099468+00:00'
[2024-06-14T19:28:39.675+0000] {redshift_data.py:122} INFO - Executing statement: 
CREATE TABLE public.yellow2_201601(
vendorid bigint, 
tpep_pickup_datetime timestamp,
tpep_dropoff_datetime timestamp, 
passenger_count double precision, 
trip_distance double precision, 
ratecodeid double precision, 
store_and_fwd_flag varchar(12), 
Pulocationid bigint, 
dolocationid bigint, 
payment_type bigint, 
fare_amount double precision, 
extra double precision, 
mta_tax double precision, 
tip_amount double precision, 
tolls_amount double precision, 
improvement_surcharge double precision, 
total_amount double precision, 
congestion_surcharge double precision, 
airport_fee double precision);
    
[2024-06-14T19:28:39.678+0000] {base.py:83} INFO - Using connection ID 'aws_default' for task execution.
[2024-06-14T19:28:39.679+0000] {connection_wrapper.py:378} INFO - AWS Connection (conn_id='aws_default', conn_type='aws') credentials retrieved from login and password.
[2024-06-14T19:28:40.647+0000] {redshift_data.py:113} INFO - Polling statement 043b9d17-8d23-4dc6-b4ba-6ae526f905c4
[2024-06-14T19:28:40.779+0000] {redshift_data.py:129} INFO - Query PICKED
[2024-06-14T19:28:50.793+0000] {redshift_data.py:113} INFO - Polling statement 043b9d17-8d23-4dc6-b4ba-6ae526f905c4
[2024-06-14T19:28:51.007+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/amazon/aws/operators/redshift_data.py", line 124, in execute
    self.statement_id = self.hook.execute_query(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/amazon/aws/hooks/redshift_data.py", line 107, in execute_query
    self.wait_for_results(statement_id, poll_interval=poll_interval)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/amazon/aws/hooks/redshift_data.py", line 124, in wait_for_results
    raise ValueError(
ValueError: Statement '043b9d17-8d23-4dc6-b4ba-6ae526f905c4' terminated with status FAILED. Response details: {'CreatedAt': datetime.datetime(2024, 6, 14, 19, 28, 40, 417000, tzinfo=tzlocal()),
 'Database': 'dev',
 'DbUser': 'IAM:velib_project',
 'Duration': -1,
 'Error': 'ERROR: Relation "yellow2_201601" already exists',
 'HasResultSet': False,
 'Id': '043b9d17-8d23-4dc6-b4ba-6ae526f905c4',
 'QueryString': '\n'
                'CREATE TABLE public.yellow2_201601(\n'
                'vendorid bigint, \n'
                'tpep_pickup_datetime timestamp,\n'
                'tpep_dropoff_datetime timestamp, \n'
                'passenger_count double precision, \n'
                'trip_distance double precision, \n'
                'ratecodeid double precision, \n'
                'store_and_fwd_flag varchar(12), \n'
                'Pulocationid bigint, \n'
                'dolocationid bigint, \n'
                'payment_type bigint, \n'
                'fare_amount double precision, \n'
                'extra double precision, \n'
                'mta_tax double precision, \n'
                'tip_amount double precision, \n'
                'tolls_amount double precision, \n'
                'improvement_surcharge double precision, \n'
                'total_amount double precision, \n'
                'congestion_surcharge double precision, \n'
                'airport_fee double precision);\n'
                '    ',
 'RedshiftPid': 1073955020,
 'RedshiftQueryId': 0,
 'ResponseMetadata': {'HTTPHeaders': {'content-length': '1020',
                                      'content-type': 'application/x-amz-json-1.1',
                                      'date': 'Fri, 14 Jun 2024 19:28:50 GMT',
                                      'x-amzn-requestid': '496d47f7-8e3d-48e0-88cc-cf3f242d4058'},
                      'HTTPStatusCode': 200,
                      'RequestId': '496d47f7-8e3d-48e0-88cc-cf3f242d4058',
                      'RetryAttempts': 0},
 'ResultRows': -1,
 'ResultSize': -1,
 'Status': 'FAILED',
 'UpdatedAt': datetime.datetime(2024, 6, 14, 19, 28, 41, 40000, tzinfo=tzlocal()),
 'WorkgroupName': 'redshift-velib'}
[2024-06-14T19:28:51.025+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=s3_to_ml_to_postgres, task_id=setup_create_table, execution_date=20240614T192722, start_date=20240614T192839, end_date=20240614T192851
[2024-06-14T19:28:51.038+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 25 for task setup_create_table (Statement '043b9d17-8d23-4dc6-b4ba-6ae526f905c4' terminated with status FAILED. Response details: {'CreatedAt': datetime.datetime(2024, 6, 14, 19, 28, 40, 417000, tzinfo=tzlocal()),
 'Database': 'dev',
 'DbUser': 'IAM:velib_project',
 'Duration': -1,
 'Error': 'ERROR: Relation "yellow2_201601" already exists',
 'HasResultSet': False,
 'Id': '043b9d17-8d23-4dc6-b4ba-6ae526f905c4',
 'QueryString': '\n'
                'CREATE TABLE public.yellow2_201601(\n'
                'vendorid bigint, \n'
                'tpep_pickup_datetime timestamp,\n'
                'tpep_dropoff_datetime timestamp, \n'
                'passenger_count double precision, \n'
                'trip_distance double precision, \n'
                'ratecodeid double precision, \n'
                'store_and_fwd_flag varchar(12), \n'
                'Pulocationid bigint, \n'
                'dolocationid bigint, \n'
                'payment_type bigint, \n'
                'fare_amount double precision, \n'
                'extra double precision, \n'
                'mta_tax double precision, \n'
                'tip_amount double precision, \n'
                'tolls_amount double precision, \n'
                'improvement_surcharge double precision, \n'
                'total_amount double precision, \n'
                'congestion_surcharge double precision, \n'
                'airport_fee double precision);\n'
                '    ',
 'RedshiftPid': 1073955020,
 'RedshiftQueryId': 0,
 'ResponseMetadata': {'HTTPHeaders': {'content-length': '1020',
                                      'content-type': 'application/x-amz-json-1.1',
                                      'date': 'Fri, 14 Jun 2024 19:28:50 GMT',
                                      'x-amzn-requestid': '496d47f7-8e3d-48e0-88cc-cf3f242d4058'},
                      'HTTPStatusCode': 200,
                      'RequestId': '496d47f7-8e3d-48e0-88cc-cf3f242d4058',
                      'RetryAttempts': 0},
 'ResultRows': -1,
 'ResultSize': -1,
 'Status': 'FAILED',
 'UpdatedAt': datetime.datetime(2024, 6, 14, 19, 28, 41, 40000, tzinfo=tzlocal()),
 'WorkgroupName': 'redshift-velib'}; 963)
[2024-06-14T19:28:51.081+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2024-06-14T19:28:51.094+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
